File: src/test/java/com/timepath/DiffTest.java
package com.timepath;

import org.jetbrains.annotations.NotNull;
import org.junit.Test;

import java.util.Arrays;
import java.util.Comparator;
import java.util.List;
import java.util.logging.Logger;

import static org.junit.Assert.assertEquals;

/**
 * @author TimePath
 */
public class DiffTest {

    private static final Logger LOG = Logger.getLogger(DiffTest.class.getName());

    /**
     * Test of diff method, of class Diff.
     */
    @Test
    @SuppressWarnings({"rawtypes", "unchecked"})
    public void testDiff() {
        System.out.println("diff");
        @NotNull Diff expResult = new Diff();
        expResult.removed = Arrays.asList("B");
        expResult.added = Arrays.asList("C");
        expResult.modified = Arrays.asList();
        expResult.same = Arrays.asList("A");
        @NotNull Comparator<String> caseInsensitive = new Comparator<String>() {
            public int compare(@NotNull String o1, @NotNull String o2) {
                return o1.toLowerCase().compareTo(o2.toLowerCase());
            }
        };
        @NotNull List<String> original = Arrays.asList("A", "B");
        @NotNull List<String> changed = Arrays.asList("A", "C");
        @NotNull Diff<String> result = Diff.diff(original, changed, caseInsensitive, null);
        System.out.println("Deleted: " + result.removed + " vs " + expResult.removed);
        System.out.println("New: " + result.added + " vs " + expResult.added);
        System.out.println("Modified: " + result.modified + " vs " + expResult.modified);
        System.out.println("Same: " + result.same + " vs " + expResult.same);
        assertEquals(expResult, result);
    }
}

File: src/test/kotlin/com/timepath/DiffTest.kt
package com.timepath

import org.junit.Assert.assertEquals
import java.util.Comparator
import org.junit.Test as test

/**
 * @author TimePath
 */
public class DiffTest {

    /**
     * Test of diff method, of class Diff.
     */
    test fun testDiff() {
        val expResult = Diff("A" to "A", listOf("C"), listOf("B"), listOf("A"), listOf())
        val caseInsensitive = Comparator<String> { o1, o2 -> o1.compareTo(o2, ignoreCase = true) }
        val original = listOf("A", "B")
        val changed = listOf("A", "C")
        val result = Diff.diff(original, changed, caseInsensitive)
        println("Deleted: " + result.removed + " vs " + expResult.removed)
        println("New: " + result.added + " vs " + expResult.added)
        println("Modified: " + result.modified + " vs " + expResult.modified)
        println("Same: " + result.same + " vs " + expResult.same)
        assertEquals(expResult, result)
    }
}
